common/persistence/history_manager.go

```diff
@@ -160,9 +160,9 @@ func (m *executionManagerImpl) DeleteHistoryBranch(
 	})
 
 	// Get the entire history tree, so we know if any part of the target branch is referenced by other branches.
```

Comment:
```suggestion
	// Get all history trees, so we know if any part of the target branch is referenced by other branches.
```

Reply:
Only one tree will contain the branch, so getting all trees is not accurate here.
I changed to
```
	// Get the history tree containing the branch to be delelted,
	// so we know if any part of the target branch is referenced by other branches.
```